config.go

```diff
@@ -50,6 +50,22 @@ func DefaultConfig(authToken string) ClientConfig {
 	}
 }
 
+func DefaultConfigWithUrl(authToken string, proxyUrl string) ClientConfig {
+	if proxyUrl == "" {
+		proxyUrl = openaiAPIURLv1
+	}
+	return ClientConfig{
+		authToken: authToken,
+		BaseURL:   proxyUrl,
+		APIType:   APITypeOpenAI,
+		OrgID:     "",
+
+		HTTPClient: &http.Client{},
+
+		EmptyMessagesLimit: defaultEmptyMessagesLimit,
+	}
+}
+
```

Comment:
FIY We can already.
```
cfg := openai.DefaultConfig("openai_api_key")
cfg.BaseURL = "proxyUrl"
```

Reply:
cool, looks like i made a wrong attempt earlier, thanks a lot.